cabal-install/Distribution/Client/ProjectOrchestration.hs

```diff
@@ -466,16 +472,135 @@ printPlan verbosity
+    --  - then we do not report additional error detail or context.
+    --
+    isSimpleCase
+      | [(pkgid, reason)] <- failures
+      , [pkg]             <- rootpkgs
+      , installedUnitId pkg == pkgid
+      , isFailureSelfExplanatory reason
+      = True
+      | otherwise
+      = False
+
+    isFailureSelfExplanatory (BuildFailed e)
+      | Just (ExitFailure _) <- fromException e = True
+
+    isFailureSelfExplanatory (ConfigureFailed e)
+      | Just (ExitFailure _) <- fromException e = True
+
+    isFailureSelfExplanatory _                  = False
+
+    rootpkgs =
+      [ pkg
+      | InstallPlan.Configured pkg <- InstallPlan.toList plan
+      , hasNoDependents pkg ]
+
+    ultimateDeps pkgid =
+        filter (\pkg -> hasNoDependents pkg && installedUnitId pkg /= pkgid)
+               (InstallPlan.reverseDependencyClosure plan [pkgid])
+
+    hasNoDependents :: HasUnitId pkg => pkg -> Bool
+    hasNoDependents = null . InstallPlan.revDirectDeps plan . installedUnitId
+
+    renderFailure pkg reason =
+        case reason of
+          DownloadFailed  e -> "failed to download " ++ pkgstr ++ "."
+                            ++ showException e
+          UnpackFailed    e -> "failed to unpack " ++ pkgstr ++ "."
+                            ++ showException e
+          ConfigureFailed e -> "failed to build " ++ pkgstr ++ ". The failure"
+                            ++ "occurred during the configure step."
+                            ++ showException e
+          BuildFailed     e -> "failed to build " ++ pkgstr ++ "."
+                            ++ showException e
+          TestsFailed     e -> "tests failed for " ++ pkgstr ++ "."
+                            ++ showException e
+          InstallFailed   e -> "failed to build " ++ pkgstr ++ ". The failure"
+                            ++ "occurred during the final install step."
+                            ++ showException e
+
+          -- This will never happen, but we include it for completeness
+          DependentFailed pkgid -> " depends on " ++ display pkgid
+                                ++ " which failed to install."
+          PlanningFailed -> " failed during the planning phase."
+      where
+        pkgstr = display (packageId pkg)
+              ++ renderDependencyOf (installedUnitId pkg)
+
+    renderDependencyOf pkgid =
+      case ultimateDeps pkgid of
+        []         -> ""
+        (p1:[])    -> " (which is required by " ++ display (packageName p1) ++ ")"
+        (p1:p2:[]) -> " (which is required by " ++ display (packageName p1)
+                                     ++ " and " ++ display (packageName p2) ++ ")"
+        (p1:p2:_)  -> " (which is required by " ++ display (packageName p1)
+                                        ++ ", " ++ display (packageName p2)
+                                        ++ " and others)"
+
+    showException e = case fromException e of
+      Just (ExitFailure 1) -> ""
+        --TODO: show log in this case
```

Comment:
This seems like a really major omission. If I'm doing a parallel build and one of the dependents fails to build, the log will not have already been printed, and this is our last chance to show the log, right?